libraries/CMakeLists.txt

```diff
@@ -1,3 +1,9 @@
 install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/
         DESTINATION "${MATERIALX_INSTALL_STDLIB_PATH}" MESSAGE_NEVER
-        PATTERN "CMakeLists.txt" EXCLUDE)
+        PATTERN "CMakeLists.txt" EXCLUDE
+        PATTERN "pbrlib_genosl_impl.legacy" EXCLUDE)
+
+if (MATERIALX_OSL_LEGACY_CLOSURES)
+    install(FILES "${CMAKE_CURRENT_SOURCE_DIR}/pbrlib/genosl/pbrlib_genosl_impl.legacy"
+        DESTINATION "${MATERIALX_INSTALL_STDLIB_PATH}/pbrlib/genosl/" RENAME pbrlib_genosl_impl.mtlx)
+endif()
```

Comment:
Some of the code snippets are different when using the legacy OSL closures. This will install a separate mtlx file with implementation elements pointing to the right code.